contracts/forwarder/Forwarder.sol

```diff
@@ -59,8 +59,10 @@ contract Forwarder is IForwarder {
         _verifySig(req, domainSeparator, requestTypeHash, suffixData, sig);
         _updateNonce(req);
 
+        bytes memory callData = abi.encodePacked(req.data, req.from);
+        require( gasleft()*63/64 >= req.gas, "FWD: insufficient gas" );
```

Comment:
in the Hub we have a `gas reserve`, I think we might want to consider it for Forwarder as well if we cannot be sure that 1/64 of gas limit will be enough to make a `transfer` and a `return` .

Reply:
we don't really care if for transfer and return succeed - it will revert entire outder transaction.
The attack is only related to failure INSIDE the addr.call(), where only inner transaction is reverted.

Reply:
Note that the `should revert if not given enough gas` tests validates that even 1 gas too little causes a revert of the outer transaction